app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -240,18 +236,10 @@ class MailHelper
      */
     private $embedImagesReplaces = [];
 
-    public function __construct(MauticFactory $factory, \Swift_Mailer $mailer, $from = null)
+    public function __construct(MauticFactory $factory, MailerInterface $mailer, $from = null)
     {
-        $this->factory   = $factory;
-        $this->mailer    = $mailer;
-        $this->transport = $mailer->getTransport();
-
-        try {
-            $this->logger = new \Swift_Plugins_Loggers_ArrayLogger();
-            $this->mailer->registerPlugin(new \Swift_Plugins_LoggerPlugin($this->logger));
-        } catch (\Exception $e) {
-            $this->logError($e);
-        }
+        $this->factory = $factory;
```

Comment:
It would be nice to remove usage of MauticFactory from `MailHelper` in this enormous refactoring. Would it be hard to do?

Reply:
I do not think it is a very hard thing to do since it is only only used to get parameters from the config file

Reply:
Based on a conversation with @escopecz , we will keep this for future refactoring